ultravox/inference/infer.py

```diff
@@ -38,6 +38,27 @@ def __init__(
             None
         )
 
+    def batch_infer(
+        self,
+        samples: List[datasets.VoiceSample],
+        max_tokens: Optional[int] = None,
+        temperature: Optional[float] = None,
+    ) -> base.InferenceGenerator:
+        inputs = [self._dataproc(s, batch=True) for s in samples]
```

Comment:
You need to enforce `tokenizer.padding_side = "left"` somewhere, most likely in the `__init__` function.

Reply:
For my understanding, is this only necessary for batch infer? Or should we have done this for regular infer as well.

Reply:
Correct. For regular infer, there isn't any padding being applied anywhere, so it doesn't matter.

For batched generation however, the sizes of the input_ids will likely differ, hence the need for padding.
We want to be able to add new (generated) tokens on the right, and that won't work correctly if we've already padded the input on the right side.

Reply:
Ah thats right, thanks for clarifying